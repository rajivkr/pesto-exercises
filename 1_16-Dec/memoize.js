const fact = () => {
  let factorialCache = {};
  return (factRecursive = n => {
    if (factorialCache[n] !== undefined) {
      console.log('Factorial fetched from factorialCache');
      return factorialCache[n];
    } else {
      let result = n == 0 ? 1 : n * factRecursive(n - 1, factorialCache);
      factorialCache[n] = result;
      return result;
    }
  });
};

let factMemo = fact();
console.log(factMemo(3));
console.log(factMemo(4));
console.log(factMemo(3));
