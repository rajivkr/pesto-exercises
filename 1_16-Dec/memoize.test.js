import { factMemo as memoize } from './memoize';

describe('using memoize test cases', () => {
  it('should use it from cache', () => {
    let res = `${memoize(5)} + ${memoize(6)} + ${memoize(5)}`;
    expect(res).toMatchSnapshot();
  });
});
