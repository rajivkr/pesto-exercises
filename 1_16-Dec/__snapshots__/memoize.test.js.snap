// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using memoize test cases should use it from cache 1`] = `"120 + 720 + 120"`;
